src/typescript/azure-openapi-validator/rules/OperationIdRequired.ts

```diff
@@ -0,0 +1,30 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { MergeStates, OpenApiTypes, rules } from '../rule';
+import { IsValidOperation } from './utilities/rules-helper';
+export const OperationIdRequired: string = "OperationIdRequired";
+
+rules.push({
+  id: "R4004",
+  name: OperationIdRequired,
+  severity: "warning",
+  category: "ARMViolation",
+  mergeState: MergeStates.individual,
+  openapiType: OpenApiTypes.arm,
+  appliesTo_JsonQuery: "$.paths",
+  run: function* (doc, node, path) {
+    if (node !== undefined) {
+      for (const pathKey in node)
+        for (const op in node[pathKey]) {
+          if (false === IsValidOperation(op)) {
+            continue;
+          }
+          if (node[pathKey][op].operationId === undefined || node[pathKey][op].operationId === "") {
+            yield { message: `Missing operationId. path:'${pathKey}', operation:'${op}'`, location: path };
```

Comment:
Could you consider using a more user friendly message?

Reply:
good question, but this message was related to the lint doc, and the doc have been approved

Reply:
You may consider updating the spec as well. Please take a look at the existing messages and follow the same pattern.

Reply:
updated